src/warc2zim/icon_finder.py

```diff
@@ -0,0 +1,84 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+
+from bs4 import BeautifulSoup, Tag
+
+ZIM_ILLUSTRATION_SIZE = 48
+
+
+@dataclass(frozen=True)
+class Icon:
+    """Helper class to compare icons for warc2zim usage"""
+
+    url: str
+    width: int
+    height: int
+    icon: bytes
+    format: str | None
+
+    def __lt__(self, other):
+        return Icon._cmp_icons(self, other) < 0
+
+    def __gt__(self, other):
+        return Icon._cmp_icons(self, other) > 0
+
+    @classmethod
+    def _cmp_icons(cls, x: "Icon", y: "Icon") -> int:
+        # icons are supposed to be squared ; should they not be squared, we consider
+        # only the smallest dimension for comparison
+        x_dim = min(x.width, x.height)
+        y_dim = min(y.width, y.height)
+        if x_dim == ZIM_ILLUSTRATION_SIZE:
+            if y_dim != ZIM_ILLUSTRATION_SIZE:
+                return 1  # x is perfect, y not
+            return 0  # both images are perfect
+        if x_dim > ZIM_ILLUSTRATION_SIZE:
+            if y_dim == ZIM_ILLUSTRATION_SIZE:
+                return -1  # y is perfect, x not
+            elif x_dim < y_dim:
+                return -1  # prefer biggest icon (y)
+            elif x_dim == y_dim:
+                return 0  # both icon are equivalent
+            else:
+                return 1  #  perfer biggest icon (x)
+        if y_dim >= ZIM_ILLUSTRATION_SIZE:  # and x < ZIM_ILLUSTRATION_SIZE
```

Comment:
```suggestion
        if y_dim >= ZIM_ILLUSTRATION_SIZE:  # x is < ZIM_ILLUSTRATION_SIZE
```
Just formatting it for its intent (reminding status) and not confuse it with a commented out condition part

Reply:
Yep